webapp/src/ts/modules/contacts/contacts.component.ts

```diff
@@ -0,0 +1,361 @@
+import { SimprintsService } from '@mm-services/simprints.service';
+import { Selectors } from '@mm-selectors/index';
+import { isMobile } from '@mm-providers/responsive.provider';
+import { SearchService } from '@mm-services/search.service';
+import { ContactTypesService } from '@mm-services/contact-types.service';
+import { ScrollLoaderProvider } from '@mm-providers/scroll-loader.provider';
+
+const PAGE_SIZE = 50;
+
+@Component({
+  templateUrl: './contacts.component.html'
+})
+export class ContactsComponent implements OnInit, OnDestroy{
+  private subscription: Subscription = new Subscription();
+  private globalActions;
+  private contactsActions;
+  private servicesActions;
+  private listContains;
+
+  contactsList;
+  loading = false;
+  error;
+  appending;
+  filtered = false;
+  hasContacts = true;
+  filters:any = {};
+  defaultFilters:any = {};
+  moreItems;
+  usersHomePlace;
+  contactTypes;
+  isAdmin;
+  childPlaces;
+  lastVisitedDateExtras;
+  visitCountSettings;
+  defaultSortDirection = 'alpha';
+  sortDirection = this.defaultSortDirection;
+  additionalListItem = false;
+  simprintsEnabled;
+  enketoEdited;
+
+  constructor(
+    private store: Store,
+    private route: ActivatedRoute,
+    private changesService: ChangesService,
+    private translateService: TranslateService,
+    private searchService: SearchService,
+    private contactTypesService: ContactTypesService,
+    private userSettingsService:UserSettingsService,
+    private getDataRecordsService: GetDataRecordsService,
+    private sessionService: SessionService,
+    private authService: AuthService,
+    private settingsService: SettingsService,
+    private UHCSettings: UHCSettingsService,
+    private simprintsService: SimprintsService,
+    private scrollLoaderProvider: ScrollLoaderProvider,
+  ) {
+    this.globalActions = new GlobalActions(store);
+    this.contactsActions = new ContactsActions(store);
+    this.servicesActions = new ServicesActions(store);
+  }
+
+  ngOnInit() {
+    const reduxSubscription = combineLatest(
+      this.store.select(Selectors.getContactsList),
+      this.store.select(Selectors.getIsAdmin),
+      this.store.select(Selectors.getFilters),
+      this.store.select(Selectors.contactListContains)
+    ).subscribe(([contactsList, isAdmin, filters, listContains]) => {
+      this.contactsList = contactsList;
+      this.isAdmin = isAdmin;
+      this.filters = filters;
+      this.listContains = listContains;
+    });
+    this.subscription.add(reduxSubscription);
+
+    const changesSubscription = this.changesService.subscribe({
+      key: 'contacts-list',
+      callback: (change) => {
+        if (change.deleted) {
+          // TODO: implement deletion logic
+        }
+        if (change.doc) {}
+        const withIds =
+          this.isSortedByLastVisited() &&
+          !!this.isRelevantVisitReport(change.doc) &&
+          !change.deleted;
+        return this.query({
```

Comment:
I don't think this is working. I'm updating a document I see in the list (changing the name) but it has no effect on the list whatsoever.